samples/sample_common/include/avc_headers.h

```diff
@@ -49,11 +49,7 @@ class HeaderSet
             m_header.resize(id + 1);
         }
 
-        if (m_header[id])
-        {
-            delete m_header[id];
-            m_header[id]=0;
-        }
+        delete m_header[id];
```

Comment:
@mgonchar : why we don't manage this memory via std::unique_ptr? Few other places in the implementation of this stuff raises questions, for example RefCounter class here: https://github.com/Intel-Media-SDK/MediaSDK/blob/master/samples/sample_common/include/avc_structures.h#L223

Reply:
@dvrogozh ```unique_ptr``` is always better for memory management) but it should be subject of another PR.

yes, code you referred to is looking as self invented ```std::shared_ptr```

Reply:
Let's track and address this in #1739.